File: src/main/java/com/protryon/jasm/MethodHandle.java
package com.protryon.jasm;

public class MethodHandle {

    public final Constant ref;
    public final MethodHandleType refType;

    public MethodHandle(Constant ref, MethodHandleType refType) {
        this.ref = ref;
        this.refType = refType;
    }

    public enum MethodHandleType {
        NULL,
        REF_getField,
        REF_getStatic,
        REF_putField,
        REF_putStatic,
        REF_invokeVirtual,
        REF_invokeStatic,
        REF_invokeSpecial,
        REF_newInvokeSpecial,
        REF_invokeInterface,
    }

}

File: src/main/java/com/protryon/jasm/MethodHandle.kt
package com.protryon.jasm

class MethodHandle(val ref: Constant<*>, val refType: MethodHandleType) {

    enum class MethodHandleType {
        NULL,
        REF_getField,
        REF_getStatic,
        REF_putField,
        REF_putStatic,
        REF_invokeVirtual,
        REF_invokeStatic,
        REF_invokeSpecial,
        REF_newInvokeSpecial,
        REF_invokeInterface
    }

}
